sunpy/coordinates/frames.py

```diff
@@ -587,6 +587,15 @@ def make_3d(self):
 
             d = np.fmin(d, dd) if self._spherical_screen['only_off_disk'] else dd
 
+        # If we have all NaNs in d, warn user
+        if np.all(np.isnan(d)):
+            raise SunpyUserWarning("The coordinate array produced by this "
+                                   "method appears to be all NaN. This may "
+                                   "occur when a non-trivial coordinate "
+                                   "transformation is requested using "
+                                   "`Helioprojective.make_3d()`: try using "
+                                   "`assume_spherical_screen()` instead.")
```

Comment:
```suggestion
            warn_user("The conversion of these 2D helioprojective coordinates to 3D is all NaNs "
                      "because off-disk coordinates need an additional assumption to be mapped to "
                      "3D. Consider using the context manager "
                      "`Helioprojective.assume_spherical_screen()`.")
```
You shouldn't use `raise` for warnings (which is a mistake I've made as well).  We have a helper function (`warn_user()`), which is imported from `sunpy.util.exceptions`.